cli/cmd/check.go

```diff
@@ -149,6 +152,8 @@ func configureAndRunChecks(wout io.Writer, werr io.Writer, stage string, options
 		checks = append(checks, healthcheck.LinkerdPreInstallChecks)
 		if !options.cniEnabled {
 			checks = append(checks, healthcheck.LinkerdPreInstallCapabilityChecks)
+		} else {
+			checks = append(checks, healthcheck.LinkerdCniPluginChecks)
```

Comment:
Ok, after taking a step back, searching my soul, communing with nature, and going on a vision quest, I think I know how to simplify this.  Behold.

The `checkClusterRoles` and related methods which are part of `LinkerdPreInstallCapabilityChecks` should use `linkerd.io/control-plane-ns,!linkerd.io/cni-resource` as the label selector to select all linkerd components except the CNI ones.  This should not require checking `noInitContainer` at all since this check ONLY cares about the regular resources, not the CNI ones.  This check is run by `LinkerdPreInstallCapabilityChecks` with shouldExist=false and `LinkerdConfigChecks` with shouldExist=true.  These checks should always run, regardless of if CNI is enabled or disabled.

We should add new `checkCniRoles` and related methods.  These methods should be checked if check is run in `--pre` mode with the CNI flag enabled and also should be checked if check is run in non-pre mode and CNI mode is marked as enabled in the linkerd config.  Specifying the CNI enabled flag to a value that differs from the value in the configmap` should be a check error.  These methods should use the `linkerd.io/cni-resource` label selector to only look at CNI resources and validate that they exist.

Reply:
We should be able to specify multiple labels in the `ListOptions.LabelSelector` where each label is separated by `,`.  See [here](https://godoc.org/k8s.io/apimachinery/pkg/labels#Parse) for the supported string format. Then we won't need all the `noInitContainer` and `skipCNI` `if` conditionals.

Reply:
@adleong 
> Specifying the CNI enabled flag to a value that differs from the value in the configmapshould be a check error.

However specifying the value to false is the same as not specifying it at all. So should we change that bool to be a pointer to a bool (feels strange..) to cover the case when it was not set at all. Alternatively we can resort to the validation that I already have in place (the one you originally suggested) that disallows setting the `--linkerd-cni-enabled` unless you are doing a `--pre` check